pkg/reconciler/taskrun/metrics.go

```diff
@@ -0,0 +1,271 @@
+	if err != nil {
+		return nil, err
+	}
+	r.status = status
+
+	pipeline, err := tag.NewKey("pipeline")
+	if err != nil {
+		return nil, err
+	}
+	r.pipeline = pipeline
+
+	pipelineRun, err := tag.NewKey("pipelinerun")
+	if err != nil {
+		return nil, err
+	}
+	r.pipelineRun = pipelineRun
+
+	pod, err := tag.NewKey("pod")
+	if err != nil {
+		return nil, err
+	}
+	r.pod = pod
+
+	err = view.Register(
+		&view.View{
+			Description: trDuration.Description(),
+			Measure:     trDuration,
+			Aggregation: trDistribution,
+			TagKeys:     []tag.Key{r.task, r.taskRun, r.namespace, r.status},
+		},
+		&view.View{
+			Description: prTRDuration.Description(),
+			Measure:     prTRDuration,
+			Aggregation: prTRLatencyDistribution,
+			TagKeys:     []tag.Key{r.task, r.taskRun, r.namespace, r.status, r.pipeline, r.pipelineRun},
+		},
+		&view.View{
+			Description: trCount.Description(),
+			Measure:     trCount,
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{r.status},
+		},
+		&view.View{
+			Description: runningTRsCount.Description(),
+			Measure:     runningTRsCount,
+			Aggregation: view.LastValue(),
+		},
+		&view.View{
+			Description: podLatency.Description(),
+			Measure:     podLatency,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{r.task, r.taskRun, r.namespace, r.pod},
+		},
+	)
+
+	if err != nil {
+		r.initialized = false
+		return r, err
+	}
+
+	return r, nil
+}
+
+func (r *Recorder) DurationAndCount(tr *v1alpha1.TaskRun) error {
+	if !r.initialized {
+		return fmt.Errorf("ignoring the metrics recording for %s , failed to initialize the metrics recorder", tr.Name)
+	}
+
+	duration := time.Since(tr.Status.StartTime.Time)
+	if tr.Status.CompletionTime != nil {
+		duration = tr.Status.CompletionTime.Sub(tr.Status.StartTime.Time)
+	}
+
+	taskName := "annonymous"
+	if tr.Spec.TaskRef != nil {
+		taskName = tr.Spec.TaskRef.Name
+	}
+
+	status := "success"
+	if tr.Status.Conditions[0].Status == corev1.ConditionFalse {
+		status = "failed"
+	}
+
+	if ok, pipelineData := tr.IsOfAPipeline(); ok {
+		ctx, err := tag.New(
+			context.Background(),
+			tag.Insert(r.task, taskName),
+			tag.Insert(r.taskRun, tr.Name),
+			tag.Insert(r.namespace, tr.Namespace),
+			tag.Insert(r.status, status),
+			tag.Insert(r.pipeline, pipelineData[pipeline.PipelineLabelKey]),
+			tag.Insert(r.pipelineRun, pipelineData[pipeline.PipelineRunLabelKey]),
+		)
+
+		if err != nil {
+			return err
+		}
+
+		stats.Record(ctx, prTRDuration.M(float64(duration/time.Second)))
+		metrics.Record(ctx, trCount.M(1))
+		return nil
+	}
+
+	ctx, err := tag.New(
+		context.Background(),
+		tag.Insert(r.task, taskName),
+		tag.Insert(r.taskRun, tr.Name),
+		tag.Insert(r.namespace, tr.Namespace),
+		tag.Insert(r.status, status),
+	)
+	if err != nil {
+		return err
+	}
+
+	metrics.Record(ctx, trDuration.M(float64(duration/time.Second)))
+	metrics.Record(ctx, trCount.M(1))
+
+	return nil
+}
+
+func (r *Recorder) RunningTrsCount(lister listers.TaskRunLister) error {
```

Comment:
`RunningTRsCount`

Or just `RunningTaskRuns` and `RunningPipelineRuns`?

Reply:
:+1: